### ISSUE TITLE
End handlers not being removed after next resolve

### ISSUE THREAD

**Issue by AKST** [basicdays/node-stream-to-async-iterator#5]
In the notReadable state it's listening for either the end or read callback to finish, unfortunately if the read callback fires it doesn't clean up the end callback. And after 11 calls to next there are 10 event listeners for the end callback.

I know in my node version I get warned about possible memory leaks.

--------

## Suggestion

The [existing logic][existing-logic] here

```js
await Promise.race([this._untilReadable(), this._untilEnd()]);
```

Could possibly changed to this

```js
const { promise: readPromise, cleanup: cleanRead } = this._untilReadable();
const { promise: endPromise,  cleanup: cleanEnd } = this._untilEnd();
await Promise.race([readPromise, endPromise]);
cleanRead();
cleanEnd();
```

The cleanUp callbacks will unhook any dangling event listeners

[existing-logic]: https://github.com/basicdays/node-stream-to-async-iterator/blob/bbedfd3f8bdd164c80b4bd70fb1d1994904e851d/lib/stream-to-async-iterator.js#L110

**Comment by basicdays** [comment]
I haven't released this fully yet, but I did publish a beta version of this that also fixes a couple other issues. If you would like to test it out, it's available under `stream-to-async-iterator@beta`.

**Comment by basicdays** [comment]
Released in v1.0.0
